app/main.py

```diff
@@ -1,8 +1,45 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self,
+                 comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def calculate_washing_price(self, car: Car) -> float:
+        return round(((self.clean_power - car.clean_mark)
+                      * car.comfort_class
+                      * self.average_rating)
+                     / self.distance_from_city_center, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        car.clean_mark = self.clean_power
+
+    def rate_service(self, rate: int) -> None:
+        total_rating_points = (self.average_rating
+                               * self.count_of_ratings
+                               + rate)
+        self.count_of_ratings += 1
+        self.average_rating = round(total_rating_points
+                                    / self.count_of_ratings, 1)
+
+    def serve_cars(self, cars: [Car]) -> float:
+        total_income = 0.0
+        for car in cars:
+            if self.clean_power > car.clean_mark:
+                total_income += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+        return round(total_income, 1)
```

Comment:
It would be beneficial to check if the `cars` list is not empty before proceeding with the loop in the `serve_cars` method to avoid unnecessary operations.